### ISSUE TITLE
Add Traverse operation to tree data structures

### ISSUE THREAD

**Issue by JMCanning78** [JMCanning78/datastructures-visualization#108]
The traverse operation is similar across the binary trees and the 2-3-4 tree.  The three traversal orders: pre-, in-, and post- are implemented similarly with recursive procedures.  In this issue, build the common infrastructure to show the traversal with the various call environments showing their local variables pointing to different parts of the tree.  The current call environment should have some color associated with it while the calls lower on the call stack are grayed out (but still visible).

Implement the Traverse function for the BinaryTree and AVLTree visualizations.  The Tree234 variation has to handle up to 4 child nodes, so it may need to be built in a separate Issue.  The Quadtree module will also have a multi-child traversal, but I don't think it can share much in functionality with the binary tree.  

This Issue depends on Issue #105.

**Comment by JMCanning78** [comment]
PR #201 adds a traverse operation for binary trees.  I'm not sure how much can be carried over into the AVLTree visualization, but it certainly provides a framework.

**Comment by JMCanning78** [comment]
The traverse operation in BinaryTreeBase works for both BinaryTree and AVLTree.  The 2-3-4 tree will need a separate method.
